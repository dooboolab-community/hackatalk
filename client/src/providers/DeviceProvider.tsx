import * as Device from 'expo-device';

import React, {FC, useEffect, useState} from 'react';

import createCtx from '../utils/createCtx';

interface Context {
  deviceType: Device.DeviceType;
  setDeviceType: React.Dispatch<React.SetStateAction<Device.DeviceType>>;
}

const [useCtx, Provider] = createCtx<Context>();

const DeviceProvider: FC = ({children}) => {
  const [deviceType, setDeviceType] = useState<Device.DeviceType>(
    Device.DeviceType.PHONE,
  );

  useEffect(() => {
    let isMounted = true;

    Device.getDeviceTypeAsync().then((result) => {
      // Only change state if this component is still mounted.
      if (isMounted) setDeviceType(result);
    });

    return () => {
      isMounted = false;
    };
  }, []);

  return (
    <Provider
      value={{
        deviceType,
        setDeviceType,
      }}>
      {children}
    </Provider>
  );
};

const DeviceContext = {
  useDeviceContext: useCtx,
  DeviceProvider,
};

export {useCtx as useDeviceContext, DeviceProvider};
export default DeviceContext;
