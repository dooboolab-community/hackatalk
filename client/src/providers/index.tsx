export * from './AuthProvider';
export * from './DeviceProvider';
export * from './ProfileModalProvider';
