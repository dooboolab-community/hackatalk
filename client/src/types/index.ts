import { StyleProp, TextStyle } from 'react-native';

import { SFC } from 'react';
import { User } from './graphql';

export enum MessageType {
  Message,
  Photo,
  File,
}

export interface Friend extends User {
  isFriend?: boolean;
  friendSince: Date;
}

interface IconProps {
  style?: StyleProp<TextStyle>;
  width?: number | string;
  height?: number | string;
  children?: never;
}

export type IconType = SFC<IconProps>;

export enum ThemeType {
  LIGHT = 'LIGHT',
  DARK = 'DARK',
}
