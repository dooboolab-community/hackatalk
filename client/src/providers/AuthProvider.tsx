import React, {useReducer} from 'react';

import {User} from '../types/graphql';
import createCtx from '../utils/createCtx';

export interface State {
  user: User | undefined;
}

interface Context {
  state: State;
  setUser(user: User | undefined): void;
}

const [useCtx, Provider] = createCtx<Context>();

// eslint-disable-next-line no-shadow
export enum ActionType {
  SetUser = 'set-user',
}

type SetUserAction = {
  type: ActionType.SetUser;
  payload: {user: User};
};
type Action = SetUserAction;

interface Props {
  children?: React.ReactElement;
  initialAuthUser?: User;
}

type Reducer = (state: State, action: Action) => State;

const setUser = (dispatch: React.Dispatch<SetUserAction>) => (
  authUser: User,
): void => {
  dispatch({
    type: ActionType.SetUser,
    payload: {user: authUser},
  });
};

const initialState: State = {
  user: undefined,
};

// eslint-disable-next-line default-param-last
const reducer: Reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SetUser:
      return {...action.payload};
    default:
      return state;
  }
};

function AuthProvider(props: Props): React.ReactElement {
  const initialAuthState = {
    user: props.initialAuthUser,
  };

  const [state, dispatch] = useReducer<Reducer>(reducer, initialAuthState);

  const actions = {
    setUser: setUser(dispatch),
  };

  return <Provider value={{state, ...actions}}>{props.children}</Provider>;
}

const AuthContext = {
  useAuthContext: useCtx,
  AuthProvider,
};

export {useCtx as useAuthContext, AuthProvider};
export default AuthContext;
