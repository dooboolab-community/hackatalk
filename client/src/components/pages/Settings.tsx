import {Button, useTheme} from 'dooboo-ui';
import React, {ReactElement} from 'react';
import {SectionList, SectionListData} from 'react-native';
import {SvgApple, SvgFacebook, SvgGoogle} from '../../utils/Icons';
import styled, {DefaultTheme} from 'styled-components/native';

import AsyncStorage from '@react-native-async-storage/async-storage';
import {FontAwesome} from '@expo/vector-icons';
import {MainStackNavigationProps} from '../navigations/MainStackNavigator';
import type {NotificationDeleteNotificationMutation} from '../../__generated__/NotificationDeleteNotificationMutation.graphql';
import {deleteNotification} from '../../relay/queries/Notification';
import {getString} from '../../../STRINGS';
import {useAuthContext} from '../../providers/AuthProvider';
import {useMutation} from 'react-relay/hooks';

const Container = styled.SafeAreaView`
  flex: 1;
  padding-top: 10px;
  background-color: ${({theme}) => theme.background};
`;

const HeaderContainer = styled.View`
  background-color: ${({theme}) => theme.background};
  border-bottom-color: ${({theme}) => theme.lineColor};
  height: 40px;
  justify-content: center;
  margin-left: 12px;
  border-bottom-width: 1px;
`;

const SectionHeader = styled.Text`
  color: ${({theme}) => theme.secondaryText};
  margin-left: 2px;
`;

const ItemContainer = styled.TouchableOpacity`
  flex-direction: row;
  padding-left: 15px;
  padding-right: 15px;
  width: 100%;
  height: 52px;
  align-items: center;
`;

const ItemLabel = styled.Text`
  color: ${({theme}) => theme.text};
  font-size: 16px;
  flex: 1;
`;

export interface Props {
  navigation: MainStackNavigationProps<'Settings'>;
}

interface SettingsOption {
  label: string;
  icon?: ReactElement;
  onPress(): void;
  testID: string;
}

function Settings(props: Props): React.ReactElement {
  let signInInfoOption: SettingsOption;

  const {setUser} = useAuthContext();
  const {theme} = useTheme();
  const {navigation} = props;

  const {
    state: {user},
  } = useAuthContext();

  const [
    commitNotification,
  ] = useMutation<NotificationDeleteNotificationMutation>(deleteNotification);

  const renderSectionItem = (
    option: SettingsOption,
    themeProps: DefaultTheme,
  ): React.ReactElement => {
    const isEmailUser = (user?.profile?.authType || 'email') === 'email';

    return (
      <ItemContainer
        onPress={isEmailUser ? option.onPress : undefined}
        testID={option.testID}>
        {option.icon || null}
        <ItemLabel style={{marginLeft: 8, marginTop: 2}}>
          {option.label}
        </ItemLabel>
        {isEmailUser ? (
          <FontAwesome name="angle-right" size={24} color={themeProps.text} />
        ) : null}
      </ItemContainer>
    );
  };

  const logout = async (): Promise<void> => {
    if (navigation) {
      AsyncStorage.removeItem('token');

      const pushToken = await AsyncStorage.getItem('push_token');

      if (pushToken) {
        const deleteNotificationMutationConfig = {
          variables: {
            token: pushToken,
          },
        };

        commitNotification(deleteNotificationMutationConfig);
      }

      setUser(undefined);
    }
  };

  switch (user?.profile?.authType) {
    case 'google':
      signInInfoOption = {
        icon: <SvgGoogle width={24} fill={theme.text} />,
        label: getString('SIGNED_IN_WITH_GOOGLE'),
        onPress: (): void => {
          navigation.navigate('ChangePw');
        },
        testID: 'change-pw-item',
      };

      break;
    case 'facebook':
      signInInfoOption = {
        icon: <SvgFacebook width={24} fill={theme.text} />,
        label: getString('SIGNED_IN_WITH_FACEBOOK'),
        onPress: (): void => {
          navigation.navigate('ChangePw');
        },
        testID: 'change-pw-item',
      };

      break;
    case 'apple':
      signInInfoOption = {
        icon: <SvgApple width={24} fill={theme.text} />,
        label: getString('SIGNED_IN_WITH_APPLE'),
        onPress: (): void => {
          navigation.navigate('ChangePw');
        },
        testID: 'change-pw-item',
      };

      break;
    case 'email':
    default:
      signInInfoOption = {
        label: getString('SIGNED_IN_WITH_EMAIL'),
        onPress: (): void => {
          navigation.navigate('ChangePw');
        },
        testID: 'change-pw-item',
      };

      break;
  }

  const settings: SectionListData<SettingsOption>[] = [
    {
      title: getString('LOGIN_INFORMATION'),
      data: [signInInfoOption],
    },
  ];

  return (
    <Container>
      <SectionList
        testID="test-section-list"
        sections={settings}
        renderItem={({item}): React.ReactElement =>
          renderSectionItem(item, theme)
        }
        keyExtractor={(item: SettingsOption): string => item.label}
        renderSectionHeader={({section: {title}}): React.ReactElement => (
          <HeaderContainer>
            <SectionHeader>{title}</SectionHeader>
          </HeaderContainer>
        )}
      />
      <Button
        testID="button-logout"
        style={{
          paddingHorizontal: 20,
          paddingVertical: 20,
        }}
        styles={{
          container: {
            width: '100%',
            height: 48,
            backgroundColor: theme.btnPrimaryLight,
            borderColor: theme.btnPrimary,
            borderWidth: 0.3,
          },
          text: {
            color: theme.btnPrimary,
            fontSize: 14,
            fontWeight: 'bold',
          },
        }}
        onPress={logout}
        text={getString('LOGOUT')}
      />
    </Container>
  );
}

export default Settings;
