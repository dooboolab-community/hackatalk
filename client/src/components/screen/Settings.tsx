import React, { ReactElement } from 'react';
import { SectionList, SectionListData } from 'react-native';
import { SvgApple, SvgFacebook, SvgGoogle } from '../../utils/Icons';
import { graphql, useMutation } from 'react-relay/hooks';
import styled, { DefaultTheme } from 'styled-components/native';

import AsyncStorage from '@react-native-async-storage/async-storage';
import { AuthType } from '../../types/graphql';
import { Button } from 'dooboo-ui';
import { FontAwesome } from '@expo/vector-icons';
import { MainStackNavigationProps } from '../navigation/MainStackNavigator';
import type { SettingsDeleteNotificationMutation } from '../../__generated__/SettingsDeleteNotificationMutation.graphql';
import { getString } from '../../../STRINGS';
import { useAuthContext } from '../../providers/AuthProvider';
import { useThemeContext } from '@dooboo-ui/theme';

const Container = styled.SafeAreaView`
  flex: 1;
  padding-top: 10px;
  background-color: ${({ theme }): string => theme.background};
`;

const HeaderContainer = styled.View`
  background-color: ${({ theme }): string => theme.background};
  border-bottom-color: ${({ theme }): string => theme.lineColor};
  height: 40px;
  justify-content: center;
  margin-left: 12px;
  border-bottom-width: 1px;
`;

const SectionHeader = styled.Text`
  color: ${({ theme }): string => theme.fontSubColor};
  margin-left: 2px;
`;

const ItemContainer = styled.TouchableOpacity`
  flex-direction: row;
  padding-left: 15px;
  padding-right: 15px;
  width: 100%;
  height: 52px;
  align-items: center;
`;

const ItemLabel = styled.Text`
  color: ${({ theme }): string => theme.fontColor};
  font-size: 16px;
  flex: 1;
`;

export interface Props {
  navigation: MainStackNavigationProps<'Settings'>;
}

const deleteNotification = graphql`
  mutation SettingsDeleteNotificationMutation($token: String!) {
    deleteNotification(token: $token) {
      id
      token
      device
      createdAt
    }
  }
`;

interface SettingsOption {
  label: string;
  icon?: ReactElement;
  onPress(): void;
  testID: string;
}

function Settings(props: Props): React.ReactElement {
  let signInInfoOption: SettingsOption;

  const { setUser } = useAuthContext();
  const { theme } = useThemeContext();
  const { navigation } = props;

  const {
    state: { user },
  } = useAuthContext();

  const [commitNotification] = useMutation<SettingsDeleteNotificationMutation>(
    deleteNotification,
  );

  const renderSectionItem = (
    option: SettingsOption,
    themeProps: DefaultTheme,
  ): React.ReactElement => {
    const isEmailUser =
      (user?.profile?.authType || AuthType.Email) === AuthType.Email;

    return (
      <ItemContainer
        onPress={isEmailUser ? option.onPress : undefined}
        testID={option.testID}
      >
        {option.icon || null}
        <ItemLabel style={{ marginLeft: 8, marginTop: 2 }}>
          {option.label}
        </ItemLabel>
        {isEmailUser ? (
          <FontAwesome
            name="angle-right"
            size={24}
            color={themeProps.fontColor}
          />
        ) : null}
      </ItemContainer>
    );
  };

  const logout = async (): Promise<void> => {
    if (navigation) {
      AsyncStorage.removeItem('token');

      const pushToken = await AsyncStorage.getItem('push_token');

      if (pushToken) {
        const deleteNotificationMutationConfig = {
          variables: {
            token: pushToken,
          },
        };

        commitNotification(deleteNotificationMutationConfig);
      }

      setUser(undefined);
    }
  };

  switch (user?.profile?.authType) {
    case AuthType.Google:
      signInInfoOption = {
        icon: <SvgGoogle width={24} fill={theme.font} />,
        label: getString('SIGNED_IN_WITH_GOOGLE'),
        onPress: (): void => {
          navigation.navigate('ChangePw');
        },
        testID: 'change-pw-item',
      };

      break;
    case AuthType.Facebook:
      signInInfoOption = {
        icon: <SvgFacebook width={24} fill={theme.font} />,
        label: getString('SIGNED_IN_WITH_FACEBOOK'),
        onPress: (): void => {
          navigation.navigate('ChangePw');
        },
        testID: 'change-pw-item',
      };

      break;
    case AuthType.Apple:
      signInInfoOption = {
        icon: <SvgApple width={24} fill={theme.font} />,
        label: getString('SIGNED_IN_WITH_APPLE'),
        onPress: (): void => {
          navigation.navigate('ChangePw');
        },
        testID: 'change-pw-item',
      };

      break;
    case AuthType.Email:
    default:
      signInInfoOption = {
        label: getString('SIGNED_IN_WITH_EMAIL'),
        onPress: (): void => {
          navigation.navigate('ChangePw');
        },
        testID: 'change-pw-item',
      };

      break;
  }

  const settings: SectionListData<SettingsOption>[] = [
    {
      title: getString('LOGIN_INFORMATION'),
      data: [signInInfoOption],
    },
  ];

  return (
    <Container>
      <SectionList
        testID="test-section-list"
        sections={settings}
        renderItem={({ item }): React.ReactElement =>
          renderSectionItem(item, theme)
        }
        keyExtractor={(item: SettingsOption): string => item.label}
        renderSectionHeader={({ section: { title } }): React.ReactElement => (
          <HeaderContainer>
            <SectionHeader>{title}</SectionHeader>
          </HeaderContainer>
        )}
      />
      <Button
        testID="button-logout"
        style={{
          root: {
            paddingHorizontal: 20,
            paddingVertical: 20,
          },
          button: {
            width: '100%',
            height: 48,
            backgroundColor: theme.btnPrimaryLight,
            borderColor: theme.btnPrimary,
            borderWidth: 0.3,
          },
          text: {
            color: theme.btnPrimary,
            fontSize: 14,
            fontWeight: 'bold',
          },
        }}
        onPress={logout}
        text={getString('LOGOUT')}
      />
    </Container>
  );
}

export default Settings;
