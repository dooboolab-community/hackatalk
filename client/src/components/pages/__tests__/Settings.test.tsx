import 'react-native';

import * as React from 'react';

import {AuthType, User} from '../../../types/graphql';
import {
  createMockNavigation,
  createTestElement,
} from '../../../../test/testUtils';
import {fireEvent, render, waitFor} from '@testing-library/react-native';

import ReactNavigation from '@react-navigation/core';
import Settings from '../Settings';
import {useAuthContext} from '../../../providers/AuthProvider';

const mockNavigation = createMockNavigation();

jest.mock('@react-navigation/core', () => ({
  ...jest.requireActual<typeof ReactNavigation>('@react-navigation/core'),
  useNavigation: () => mockNavigation,
}));

function getEmptyAuthUserWithSignInType(authType: AuthType): User {
  return {
    id: '',
    nickname: '',
    thumbURL: '',
    photoURL: '',
    statusMessage: '',
    profile: {
      authType,
      socialId: '',
    },
  };
}

describe('[Setting] screen', () => {
  it('renders without crashing with Facebook auth type', () => {
    const component = createTestElement(<Settings />, {
      user: getEmptyAuthUserWithSignInType('facebook'),
    });

    const screen = render(component);
    const json = screen.toJSON();

    expect(json).toBeTruthy();
    expect(json).toMatchSnapshot();
  });

  it('should renders without crashing with Google auth type', () => {
    const component = createTestElement(<Settings />, {
      user: getEmptyAuthUserWithSignInType('google'),
    });

    const screen = render(component);
    const json = screen.toJSON();

    expect(json).toBeTruthy();
    expect(json).toMatchSnapshot();
  });

  it('should render without crashing with Apple auth type', () => {
    const component = createTestElement(<Settings />, {
      user: getEmptyAuthUserWithSignInType('apple'),
    });

    const screen = render(component);
    const json = screen.toJSON();

    expect(json).toBeTruthy();
    expect(json).toMatchSnapshot();
  });

  describe('interactions', () => {
    it('should simulate onPress login state item', async () => {
      const component = createTestElement(<Settings />, {
        user: getEmptyAuthUserWithSignInType('email'),
      });

      const screen = render(component);
      const btn = screen.getByTestId('change-pw-item');

      fireEvent.press(btn);
      // TODO: Test what happens after pressing the button.
    });

    it('should set auth user to undefined after logout button pressed', async () => {
      // Test component for detecting changes in auth context.
      const TestComponent: React.FC<{onAuthChange: (user?: User) => void}> = ({
        onAuthChange,
      }) => {
        const {
          state: {user},
        } = useAuthContext();

        React.useEffect(() => {
          onAuthChange(user);
        }, [user, onAuthChange]);

        return <Settings />;
      };

      // This function is called whenever auth user changes.
      const mockOnAuthChange = jest.fn();

      const component = createTestElement(
        <TestComponent onAuthChange={mockOnAuthChange} />,
        {
          user: getEmptyAuthUserWithSignInType('apple'),
        },
      );

      const screen = render(component);

      fireEvent.press(screen.getByTestId('button-logout'));

      // Auth user is set to undefined after logout.
      await waitFor(() => {
        expect(mockOnAuthChange).toHaveBeenCalledWith(undefined);
      });
    });
  });
});
