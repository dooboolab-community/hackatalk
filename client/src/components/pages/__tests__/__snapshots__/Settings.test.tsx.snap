// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Setting] screen renders without crashing with Facebook auth type 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "#000",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingTop": 10,
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "data": Array [
              Object {
                "icon": <SvgMock
                  fill="white"
                  width={24}
                />,
                "label": "Signed in with Facebook",
                "onPress": [Function],
                "testID": "change-pw-item",
              },
            ],
            "title": "Login information",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={
        Array [
          0,
        ]
      }
      testID="test-section-list"
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#000",
                  "borderBottomColor": "#515151",
                  "borderBottomWidth": 1,
                  "height": 40,
                  "justifyContent": "center",
                  "marginLeft": 12,
                },
                undefined,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "marginLeft": 2,
                  },
                  undefined,
                ]
              }
            >
              Login information
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 52,
                "opacity": 1,
                "paddingLeft": 15,
                "paddingRight": 15,
                "width": "100%",
              }
            }
            testID="change-pw-item"
          >
            <SvgMock
              fill="white"
              width={24}
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontSize": 16,
                  },
                  Object {
                    "marginLeft": 8,
                    "marginTop": 2,
                  },
                ]
              }
            >
              Signed in with Facebook
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        />
      </View>
    </RCTScrollView>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 20,
        }
      }
      testID="button-logout"
    />
  </RCTSafeAreaView>
</RNCSafeAreaProvider>
`;

exports[`[Setting] screen should render without crashing with Apple auth type 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "#000",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingTop": 10,
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "data": Array [
              Object {
                "icon": <SvgMock
                  fill="white"
                  width={24}
                />,
                "label": "Signed in with Apple",
                "onPress": [Function],
                "testID": "change-pw-item",
              },
            ],
            "title": "Login information",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={
        Array [
          0,
        ]
      }
      testID="test-section-list"
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#000",
                  "borderBottomColor": "#515151",
                  "borderBottomWidth": 1,
                  "height": 40,
                  "justifyContent": "center",
                  "marginLeft": 12,
                },
                undefined,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "marginLeft": 2,
                  },
                  undefined,
                ]
              }
            >
              Login information
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 52,
                "opacity": 1,
                "paddingLeft": 15,
                "paddingRight": 15,
                "width": "100%",
              }
            }
            testID="change-pw-item"
          >
            <SvgMock
              fill="white"
              width={24}
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontSize": 16,
                  },
                  Object {
                    "marginLeft": 8,
                    "marginTop": 2,
                  },
                ]
              }
            >
              Signed in with Apple
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        />
      </View>
    </RCTScrollView>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 20,
        }
      }
      testID="button-logout"
    />
  </RCTSafeAreaView>
</RNCSafeAreaProvider>
`;

exports[`[Setting] screen should renders without crashing with Google auth type 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "#000",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingTop": 10,
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "data": Array [
              Object {
                "icon": <SvgMock
                  fill="white"
                  width={24}
                />,
                "label": "Signed in with Google",
                "onPress": [Function],
                "testID": "change-pw-item",
              },
            ],
            "title": "Login information",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={
        Array [
          0,
        ]
      }
      testID="test-section-list"
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#000",
                  "borderBottomColor": "#515151",
                  "borderBottomWidth": 1,
                  "height": 40,
                  "justifyContent": "center",
                  "marginLeft": 12,
                },
                undefined,
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "marginLeft": 2,
                  },
                  undefined,
                ]
              }
            >
              Login information
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 52,
                "opacity": 1,
                "paddingLeft": 15,
                "paddingRight": 15,
                "width": "100%",
              }
            }
            testID="change-pw-item"
          >
            <SvgMock
              fill="white"
              width={24}
            />
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontSize": 16,
                  },
                  Object {
                    "marginLeft": 8,
                    "marginTop": 2,
                  },
                ]
              }
            >
              Signed in with Google
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        />
      </View>
    </RCTScrollView>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "paddingHorizontal": 20,
          "paddingVertical": 20,
        }
      }
      testID="button-logout"
    />
  </RCTSafeAreaView>
</RNCSafeAreaProvider>
`;
